templates/commands/render/action_include.go

```diff
@@ -102,6 +110,22 @@ func includePath(ctx context.Context, inc *spec.IncludePath, sp *stepParams) err
 					}, nil
 				}
 
+				pathIncParent := filepath.Join(p.Val, relToAbsSrc)
```

Comment:
I don't think we need to compute pathIncParent. The value we want is already being computed below as `relToFromDir`, and we can simply reuse it and call `checkIgnore(sp.ignorePatterns, relToFromDir)`.

So it would look like:

```
			Visitor: func(relToAbsSrc string, de fs.DirEntry) (common.CopyHint, error) {
				if _, ok := skipNow[relToAbsSrc]; ok {
					return common.CopyHint{
						Skip: true,
					}, nil
				}

				abs := filepath.Join(absSrc, relToAbsSrc)
				relToFromDir, err := filepath.Rel(fromDir, abs)
				if err != nil {
					return common.CopyHint{}, fmt.Errorf("filepath.Rel(%s,%s)=%w", fromDir, abs, err)
				}
				matched, err := checkIgnore(sp.ignorePatterns, relToFromDir)
				if err != nil {
					return common.CopyHint{},
						fmt.Errorf("failed to match path(%q) with ignore patterns: %w", relToFromDir, err)
				}
				if matched {
					logger.DebugContext(ctx, "path ignored", "path", relToFromDir)
					return common.CopyHint{
						Skip: true,
					}, nil
				}

				if !de.IsDir() && inc.From.Val == "destination" {
					sp.includedFromDest = append(sp.includedFromDest, relToFromDir)
				}

				return common.CopyHint{
					// Allow later includes to replace earlier includes in the
					// scratch directory. This doesn't affect whether files in
					// the final *destination* directory will be overwritten;
					// that comes later.
					Overwrite: true,
				}, nil
			},
```

Reply:
Nice catch, thanks!